spanner/cloud-client/src/test/java/com/example/spanner/SpannerSampleIT.java

```diff
@@ -241,6 +252,83 @@ public void testSample() throws Exception {
     assertThat(out).contains("1 1 Total Junk");
     out = runSample("querywithqueryoptions");
     assertThat(out).contains("1 1 Total Junk");
+
+    String backupName =
+        String.format(
+            "%s_%02d",
+            dbId.getDatabase(), LocalDate.now().get(ChronoField.ALIGNED_WEEK_OF_YEAR));
+    BackupId backupId = BackupId.of(dbId.getInstanceId(), backupName);
+
+    out = runSample("createbackup");
+    assertThat(out).contains("Created backup [" + backupId + "]");
+
+    out = runSample("cancelcreatebackup");
+    assertThat(out).contains(
+        "Backup operation for [" + backupId + "_cancel] successfully cancelled");
+
+    out = runSample("listbackupoperations");
+    assertThat(out).contains(
+        String.format(
+            "Backup %s on database %s pending:",
+            backupId.getName(),
+            dbId.getName()));
+
+    out = runSample("listbackups");
+    assertThat(out).contains(databaseId);
+
+    // Try the restore operation in a retry loop, as there is a limit on the number of restore
+    // operations that is allowed to execute simultaneously, and we should retry if we hit this
+    // limit.
+    int restoreAttempts = 0;
+    while (true) {
+      try {
+        out = runSample("restorebackup");
+        assertThat(out).contains(
+            "Restored database ["
+                + dbId.getName()
+                + "] from ["
+                + backupId.getName()
+                + "]");
+        break;
+      } catch (SpannerException e) {
+        if (e.getErrorCode() == ErrorCode.FAILED_PRECONDITION
+            && e.getMessage()
+                .contains("Please retry the operation once the pending restores complete")) {
+          restoreAttempts++;
+          if (restoreAttempts == 10) {
+            System.out.println(
+                "Restore operation failed 10 times because of other pending restores. "
+                + "Giving up restore.");
+            break;
+          }
+          Uninterruptibles.sleepUninterruptibly(60L, TimeUnit.SECONDS);
+        } else {
+          throw e;
+        }
+      }
+    }
+
+    out = runSample("listdatabaseoperations");
+    assertThat(out).contains(
+        String.format(
+            "Database %s restored from backup",
+            DatabaseId.of(
+                dbId.getInstanceId(),
+                SpannerSample.createRestoredSampleDbId(dbId))
+            .getName()));
+
+    out = runSample("updatebackup");
+    assertThat(out).contains(
+        String.format("Updating expire time of backup [%s]", backupId.toString()));
```

Comment:
I think it's better to check for "Updated backup [%s]".

Reply:
Good spot, fixed.